<a href="https://colab.research.google.com/github/Frank-Howd/DS-Unit-1-Sprint-1-Data-Wrangling-and-Storytelling/blob/master/LS_DS_121_ttests_Assignment_AG.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Autograded Notebook (Canvas & CodeGrade)

This notebook will be automatically graded. It is designed to test your answers and award points for the correct answers. Following the instructions for each Task carefully.

### Instructions

* **Download this notebook** as you would any other ipynb file
* **Upload** to Google Colab or work locally (if you have that set-up)
* **Delete `raise NotImplementedError()`**
* Write your code in the `# YOUR CODE HERE` space
* **Execute** the Test cells that contain `assert` statements - these help you check your work (others contain hidden tests that will be checked when you submit through Canvas)
* **Save** your notebook when you are finished
* **Download** as a `ipynb` file (if working in Colab)
* **Upload** your complete notebook to Canvas (there will be additional instructions in Slack and/or Canvas)

# Lambda School Data Science - Unit 1 Sprint 2 Module 1

## Hypothesis Testing - One and two-sample t-Tests

### Objectives

* Explain the purpose of a t-test and identify applications
* Use a t-test for independence to test for a statistically significant association between two categorical variables
* Use a t-test p-value to draw the correct conclusion about the null and alternative hypothesis

### Introduction

Mosquito nets have traditionally been an important tool to prevent mosquito bites in parts of the world where malaria is endemic. However, it may not be practical for an army that is on the move to set up and carry mosquito nets each night and day. Impregnating soldiers’ uniforms with insect repellent solves the mobility problem but also has drawbacks. First, the insect repellent quickly becomes ineffective with repeated washing and ironing and must be frequently reapplied. Second, in hot and humid climates the insect repellent can be absorbed through the skin, and the long-term effects of this exposure are unknown. One compromise is to have soldiers apply patches treated with insect repellent to their clothing. These patches would last longer because they would not be washed or ironed and would not expose the entire body to the insect repellent.

### Dataset Description

The `Mosquito.xlsx` dataset contains data recorded in an experiment conducted on male soldiers in the Indian Army who were stationed in the Tezpur/Solmara garrison in Northeast India. Thirty soldiers were randomly selected to receive one of three types of mosquito single repellent patch. After giving informed consent, the study participants affixed the patches at predetermined points on their uniforms and research assistants (who were blinded to the type of repellent used) counted the number of times a mosquito landed on each individual in an hour. 

Medical officers with the Indian Army have recorded data on mosquito bites and related illness for many years and can say with authority that the mean number of mosquito touches for soldiers not wearing any mosquito repellent is 8.2 per hour.**We wish to determine if wearing a single repellent patch changes the mean number of mosquito touches for soldiers compared to not wearing any mosquito repellant.**

*Adapted from: A. Bhatnagar and V.K. Mehta (2007). "Efficacy of Deltamethrin and Cyfluthrin Impregnated Cloth Over Uniform Against Mosquito Bites," Medical Journal Armed Forces India, Vol. 63, pp. 120-122.*

**Task 1** - Load the data

Let's load the data! The URL has been provided as well as the imports for pandas and numpy.

* load your CSV file into a DataFrame named `df_mosquito`

In [2]:
# Task 1
import pandas as pd
import numpy as np

# URL for the dataset
data_url = 'https://raw.githubusercontent.com/LambdaSchool/data-science-practice-datasets/main/unit_1/Mosquito/Mosquito.csv'

# YOUR CODE HERE
df_mosquito = pd.read_csv(data_url)

# Print out your DataFrame
print(df_mosquito.shape)
df_mosquito.head()

(90, 2)


ID  Mosq_count
0   1           4
1   2          10
2   3          13
3   4           0
4   5          11

**Task 1 Test**

In [3]:
# Task 1 - Test

assert isinstance(df_mosquito, pd.DataFrame), 'Have you created a DataFrame named `df_mosquito`?'
assert len(df_mosquito) == 90


**Task 2 -** Statistical hypotheses

Write the null and alternative hypotheses using the experiment information described above.

This task will not be autograded - but it is part of completing the challenge.

**Task 2 ANSWER:**

FILL IN YOUR ANSWERS BELOW

Null hypothesis: 

Alternative hypothesis: 

$H_0: \mu = 8.2$ 

$H_a: \mu \neq 8.2$ 

**Task 3** - Calculate the mean

* Calculate the mean number of mosquito touches in the sample. Assign your answer to the variable `mosquito_touch_mean`.

In [10]:
# Task 3

# YOUR CODE HERE
mosquito_touch_mean = df_mosquito['Mosq_count'].mean()
mosquito_touch_mean

8.011111111111111

**Task 3 Test**

In [6]:
# Task 3 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 4** - Calculate the standard deviation

* Calculate the standard deviation of the number of mosquito touches in the sample. Assign your answer to `mosquito_touch_std`.

In [11]:
# Task 4

# YOUR CODE HERE
mosquito_touch_std = df_mosquito['Mosq_count'].std()
mosquito_touch_std

3.2825532828777257

**Task 4 Test**

In [12]:
# Task 4 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 5** - One-sample t-test

* Conduct a 1-sample t-test to test your hypotheses. Assign your t-test result to the variable `mosquito_pval`.

*Hint: The `stats.ttest_1samp()` function returns two values; read the documentation to select the p-value*

In [17]:
# Task 5

# Use the 'ttest_1samp' from the stats package
from scipy import stats 

# YOUR CODE HERE

mosquito_pval = stats.ttest_1samp(df_mosquito['Mosq_count'], 8.2)[1]
mosquito_pval

0.5864980356272131

**Task 5 Test**

In [18]:
# Task 5 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 6**

Report your conclusion at the 0.05 significance level. Write your answer in the cell below.

This task will not be autograded - but it is part of the project!

**Task 6 ANSWER:**

The outcome of the ttest is 0.5866 or 58.66%, which supports the null hypothesis (u = 8.2), and is not rejected. 

## Use the following information to complete Tasks 7-13

### Introduction

More than 14,000 people finished the 2020 Disney Marathon held on January 12. The results by age and gender group are included in the `Disney.csv dataset`. 

**We wish to determine if the mean finishing time for male and female marathon runners is the same or if there is a difference in the mean finishing time between male and female marathon runners.**


[Source: Track Shack. 2020 Disney Marathon Race Results](https://www.trackshackresults.com/disneysports/results/wdw/wdw20/mar_results.php)

**Task 7** - Load the next dataset

Let's load the data! The URL has been provided.

* load your CSV file into a DataFrame named `df_disney`

In [21]:
# Task 7

# URL for Disney marathon dataset
data_url2 = 'https://raw.githubusercontent.com/LambdaSchool/data-science-practice-datasets/main/unit_1/Disney_Marathon/Disney.csv'

# YOUR CODE HERE
df_disney = pd.read_csv(data_url2)

print(df_disney.shape)
print(df_disney['time'].mean())
df_disney.head()


(14106, 5)
6.0621262031894485


ID gender  age   group      time
0   1      M   30  M30-34  2.371944
1   2      M   26  M25-29  2.450556
2   3      M   32  M30-34  2.457778
3   4      M   35  M35-39  2.655833
4   5      M   26  M25-29  2.736111

**Task 7 Test**

In [20]:
# Task 7 - Test

assert isinstance(df_disney, pd.DataFrame), 'Have you created a DataFrame named `df_disney`?'
assert len(df_disney) == 14106


**Task 8** - Write the hypothesis

Write the null and alternative hypotheses for the scenario described above in words and symbols.

This task will not be autograded - but it is part of completing the challenge.

**Task 8 ANSWER:**

FILL IN YOUR ANSWERS BELOW

Null hypothesis: 

Alternative hypothesis: 

$H_0: \mu = 6.06$ 

$H_a: \mu \neq 6.06$ 

**Task 9** - Create new series from a DataFrame

Create two separate Series (a pandas DataFrame column is a Series):

* one containing finishing times for male participants (`male_finish`)
* one containing finishing times for female participants (`female_finish`)

In [38]:
# Task 9

# YOUR CODE HERE

condition_male = df_disney['gender'] == 'M'
condition_female = df_disney['gender'] == 'F'

male_finish = df_disney[condition_male]
female_finish = df_disney[condition_female]

male_finish.head()

male_finish = male_finish.drop(['ID', 'gender', 'age', 'group'], axis=1)
male_finish.columns = ['Male Time']
print(male_finish.head())

female_finish = female_finish.drop(['ID', 'gender', 'age', 'group'], axis=1)
female_finish.columns = ['Female Time']
print(female_finish.head())

   Male Time
0   2.371944
1   2.450556
2   2.457778
3   2.655833
4   2.736111
    Female Time
16     2.905278
27     2.985278
61     3.207778
69     3.244444
85     3.290833


**Task 9 Test**

In [39]:
# Task 9 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 10** - Calculate the mean finishing times

* Calculate the mean finishing time for male and female participants separately. Name your variables `male_finish_mean` and `female_finish_mean`.

In [40]:
# Task 10

# YOUR CODE HERE
male_finish_mean = male_finish.mean()
female_finish_mean = female_finish.mean()

print(male_finish_mean, female_finish_mean)

Male Time    5.79916
dtype: float64 Female Time    6.291842
dtype: float64


**Task 10 Test**

In [41]:
# Task 10 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 11** - Calculate the standard deviation

* Calculate standard deviation of the mean finishing time for male and female participants separately. Name your variables `male_finish_std` and `female_finish_std`.

In [42]:
# Task 11

# YOUR CODE HERE
male_finish_std = male_finish.std()
female_finish_std = female_finish.std()

print(male_finish_std, female_finish_std)

Male Time    1.100676
dtype: float64 Female Time    0.89669
dtype: float64


**Task 11 Test**

In [43]:
# Task 11 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 12** - Conduct a 2-sample t-test

Conduct a 2-sample t-test to test your hypotheses:

* Assign the t-statistic to a variable called `disney_tval`
* Assign the p-value to a variable called `disney_pval`

Hint: The function returns two values and you can assign them with one line (example):

`variable1, variable2` = `some.function(arguments)`

In [48]:
# Task 12

# YOUR CODE HERE
disney_tval, disney_pval = stats.ttest_ind(male_finish, female_finish, equal_var=True)
print(disney_tval, disney_pval)

[-29.27857394] [5.48513801e-183]


**Task 12 Test**

In [49]:
# Task 12 - Test
# Hidden tests - you will see the results when you submit to Canvas

**Task 13** - Report your result

Report your conclusion at the 0.05 significance level.

This task will not be autograded - but it is part of completing the challenge.

The p_value is significantly less than the 0.05 confidence level, and the null hypothesis can be refuted; instead we accept the alternative hypothesis - the mean finishing times for males and females are not equal.

### In your own words, answer the following questions:

These tasks will not be autograded - but they are part of completing the challenge.

**Task 14:** Explain the Central Limit Theorem.

The central limit theorom states that even if the population distribution is not normalized, the means of the samples form a t-distribution; and for sample sizes > 30, they approximate a normal distribution; from here the standard error can be calculated, and inferences may be made.

**Task 15:** Describe the Normal Distribution.

The normal distribution is a symmetric distribution where 68% of the distribution values lie 1 std deviation from the mean, and 95% lie within 2 std deviations. 

**Task 16:** Describe the relationship between the Normal distribution and the t-distribution.

The normal distribution assumes that the population standard deviation is know, but the t-distribution does not make this assumption, and is defined by the DOF.

**Task 17:** Write about who William Sealy Gosset was.

William Sealy Gosset was a chemist and brewer who served as Head Brewer of Guinness of Guinness and was a pioneer of modern statistics.


## Portfolio Project Milestone

Write the first draft of your research question.  This is a question that should be answerable with two visualizations and a blog post.  

You should have selected a dataset and have a good idea what your research question is by the end of the day.  If you don't - please ask for help from your instructor, track team or mentor.